### PR TITLE
Feat: ContributionResolver - delete mutation

### PR DESCRIPTION
**Pull request by Elweyn**
## 🍰 Pullrequest
Method to delete a contribution

### Issues

- relates #2000

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/ContributionResolver.ts [hunk 1]
```diff
@@ -32,4 +32,22 @@ export class ContributionResolver {
     await Contribution.save(contribution)
     return new UnconfirmedContribution(contribution, user, creations)
   }
+
+  @Authorized([RIGHTS.DELETE_CONTRIBUTION])
+  @Mutation(() => Boolean)
+  async adminDeleteContribution(
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionResolver.ts:38]:
```suggestion
  async deleteContribution(
```

**Review by Mogge — CHANGES_REQUESTED** [review]

**Review by Mogge — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolver/ContributionResolver.ts (hunk 1 above)
Comment (Elweyn) [reply backend/src/graphql/resolver/ContributionResolver.ts]: Change the name

File: backend/src/graphql/resolver/ContributionResolver.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { RIGHTS } from '@/auth/RIGHTS'
 import { Context, getUser } from '@/server/context'
 import { backendLogger as logger } from '@/server/logger'
 import { Contribution } from '@entity/Contribution'
-import { Args, Authorized, Ctx, Mutation, Resolver } from 'type-graphql'
+import { Arg, Args, Authorized, Ctx, Int, Mutation, Resolver } from 'type-graphql'
 import ContributionArgs from '../arg/ContributionArgs'
```
[the remark this answers is not in the document]
Comment (Elweyn) [reply backend/src/graphql/resolver/ContributionResolver.ts]: Has been resolved through merging master

**Review by Tirokk — APPROVED** [review]
Looks really good to me! 🚀💫

I approve! 🍀

File: backend/src/graphql/resolver/ContributionResolver.test.ts [hunk 3]
```diff
@@ -516,4 +522,98 @@ describe('ContributionResolver', () => {
...
+                id: result.data.createContribution.id,
+              },
+            }),
+          ).resolves.toEqual(
+            expect.objectContaining({
+              errors: [new GraphQLError('Can not delete contribution of another user')],
+            }),
+          )
+        })
+      })
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/graphql/resolver/ContributionResolver.test.ts:603]: There is a call `adminDeleteContribution` @Tirokk

File: backend/src/graphql/resolver/ContributionResolver.test.ts [hunk 4]
```diff
@@ -522,4 +528,98 @@ describe('ContributionResolver', () => {
...
+
+      describe('User deletes own contribution', () => {
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:611]: Please add a test where you confirm the contribution and check, that a confirmed contribution cannot be deleted.

File: backend/src/graphql/resolver/ContributionResolver.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Elweyn) [reply backend/src/graphql/resolver/ContributionResolver.test.ts:609]: Yes for admins we have ``adminDeleteContribution`` and for users we have ``deleteContribution``
A User can only delete his contribution, an Admin can also delete contributions of other users

File: backend/src/graphql/resolver/ContributionResolver.ts [hunk 5]
```diff
@@ -38,6 +38,24 @@ export class ContributionResolver {
...
+    @Ctx() context: Context,
+  ): Promise<boolean> {
+    const user = getUser(context)
+    const contribution = await dbContribution.findOne(id)
+    if (!contribution) {
+      throw new Error('Contribution not found for given id.')
+    }
+    if (contribution.userId !== user.id) {
+      throw new Error('Can not delete contribution of another user')
+    }
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionResolver.ts:54]:
```
if (contribution.confirmedAt) throw new Error('A confirmed contribution cannot be deleted.')

**Review by Mogge — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolver/ContributionResolver.ts (hunk 5 above)
Comment (Elweyn) [reply backend/src/graphql/resolver/ContributionResolver.ts:54]: OK I make this

**Review by Mogge — APPROVED** [review]

**Pull request merged**